### ISSUE TITLE
Allow choice cards to wrap

### ISSUE THREAD

**Issue by SiAdcock** [guardian/source#551]
**Describe the bug**

If choice cards don't fit into their container, they should wrap onto the next line

**To Reproduce**

Constrain the width of the container for a set of choice cards. Observe that the cards break out of their container

**Expected behavior**

The cards should wrap onto the next line. Currently, cards take up `1/n` of the available space, where `n` is the number of choice cards (`flex-grow: 1`). In order to preserve the widths of each choice card, we would need to switch from using flexbox to using [grid](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Mastering_Wrapping_of_Flex_Items#Single-dimensional_layout_explained) to layout the cards.

**Screenshots**

Currently:

![Screenshot 2020-10-01 at 14 48 03](https://user-images.githubusercontent.com/5931528/94818440-0a06fb80-03f6-11eb-8f20-ef26fdb9a402.png)

Ideally:

![Screenshot 2020-10-01 at 14 59 47](https://user-images.githubusercontent.com/5931528/94819091-c3fe6780-03f6-11eb-8386-4abd1d7fdd04.png)

**Comment by rrrahal** [comment]
Hello,

I was looking at this problem and I have a question.
Is there a max size that a choice card should have?
Because when we set something like: `grid-template-columns` you have to set some sort of ceiling for the grid to know when a column ends, otherwise the button will occupy the whole space available (`flex-grow: 1`) 

I got it with `grid-template-columns: repeat(auto-fill, minmax(140px, 1fr));` but I don't think that's 100% correct:


<img width="516" alt="Screen Shot 2020-10-01 at 18 31 41" src="https://user-images.githubusercontent.com/10249593/94843394-8d374a00-0414-11eb-8cad-c676e9960cec.png">
this is without constraining the container size

<img width="499" alt="Screen Shot 2020-10-01 at 18 32 06" src="https://user-images.githubusercontent.com/10249593/94843402-8f99a400-0414-11eb-9ec0-d02129bc2918.png">
this is constraining the container size

**Comment by SiAdcock** [comment]
Hi @rrrahal. Thanks so much for taking a look at this 🙏  

It's such a tricky problem! I suspect that you are correct and that there's no way of doing this without hardcoding either a ceiling or a minimum width for choice cards. However I don't think we should do this. It would be too constraining and would make these components less useful.

I have an alternative proposal that involves expanding the API.

**Default behaviour**

```tsx
<ChoiceCardGroup>
  <ChoiceCard label="Single" />
  <ChoiceCard label="Fortnightly" />
  <ChoiceCard label="Monthly" />
  <ChoiceCard label="Half Yearly" />
  <ChoiceCard label="Annually" />
</ChoiceCardGroup>
```

The current behaviour. Choice cards occupy a single row, and will shrink as much as possible before breaking out of the boundary of the parent.

**Tiling behaviour**

```tsx
<ChoiceCardGroup columns={3}>
  <ChoiceCard label="Single" />
  <ChoiceCard label="Fortnightly" />
  <ChoiceCard label="Monthly" />
  <ChoiceCard label="Half Yearly" />
  <ChoiceCard label="Annually" />
</ChoiceCardGroup>
```

Choice cards will be tiled with the specified number of `columns`. If there are more choice cards than columns, choice cards will wrap onto a new row. Choice cards will all be of equal width. This behaviour is inspired by the [Tiles component](https://seek-oss.github.io/braid-design-system/components/Tiles) in the Braid Design System.

I think this is achievable using flex, without having to resort to CSS grid. Let me know what you think

**Comment by rrrahal** [comment]
I like expanding the API.
I managed to do it but I couldn't figured it out using flex (it was the same issue involving the size of the choice card that wraps). 
Using grid on the other hand it was pretty easy, apart from some small margin changes.
I will open a PR for you to check and see if it is useful.

**Comment by SiAdcock** [comment]
Fixed by #554
